src/index.js

```diff
@@ -8,272 +8,371 @@ const _ = require("lodash");
-
-    for (let i = 0; i < libraries.length; i++) {
-      let lib = libraries[i];
-      let code = fs.readFileSync(lib.file, "utf-8");
-      let res;
-      if (libraryNameToId[lib.name]) {
-        // update library and get a new versionId
-        let id = libraryNameToId[lib.name];
-        res = await updateLibrary(id, lib.description, code, lib.language);
-        core.info(`updated library: ${lib.name}`);
-      } else {
-        // create a new library
-        res = await createLibrary(
-          lib.name,
-          lib.description,
-          code,
-          lib.language
-        );
-        core.info(`created library: ${lib.name}`);
-      }
-      libraryDict[res.data.versionId] = { ...lib, id: res.data.id };
+  core.info(`Upserting transformations`);
+
+  for (const tr of transformations) {
+    const code = fs.readFileSync(tr.file, "utf-8");
+    let res;
+    if (transformationNameToId[tr.name]) {
+      // update existing transformer and get a new versionId
+      const id = transformationNameToId[tr.name];
+      res = await updateTransformation(
+        id,
+        tr.name,
+        tr.description,
+        code,
+        tr.language
+      );
+    } else {
+      // create new transformer
+      res = await createTransformation(
+        tr.name,
+        tr.description,
+        code,
+        tr.language
+      );
     }
-    core.info("Libraries create/update done!");
-
-    let transformationTest = [];
-    let librariesTest = [];
-
-    core.info("Building test suite...");
-    for (let i = 0; i < Object.keys(transformationDict).length; i++) {
-      let trVersionId = Object.keys(transformationDict)[i];
-      let testInputPath =
-        transformationDict[trVersionId]["test-input-file"] || "";
-      let testInput = testInputPath
-        ? JSON.parse(fs.readFileSync(testInputPath))
-        : "";
-      if (testInput) {
-        transformationTest.push({ versionId: trVersionId, testInput });
-      } else {
-        core.info(
-          `No test input provided. Testing ${transformationDict[trVersionId].name} with default payload`
-        );
-        transformationTest.push({ versionId: trVersionId });
-      }
+    transformationDict[res.data.versionId] = { ...tr, id: res.data.id };
+  }
+
+  return transformationDict;
+}
+
+// Create or update a library.
+async function upsertLibraries(libraries, libraryNameToId) {
+  const libraryDict = {};
+  core.info(`Upserting libraries`);
+  for (const lib of libraries) {
+    const code = fs.readFileSync(lib.file, "utf-8");
+    let res;
+    if (libraryNameToId[lib.name]) {
+      // update library and get a new versionId
+      const id = libraryNameToId[lib.name];
+      res = await updateLibrary(id, lib.description, code, lib.language);
+      core.info(`Updated library: ${lib.name}`);
+    } else {
+      // create a new library
+      res = await createLibrary(lib.name, lib.description, code, lib.language);
+      core.info(`Created library: ${lib.name}`);
```

Comment:
Either we go for logs before the call or after it, since we are doing before the call, let's print:
```suggestion
      core.info(`Creating library: ${lib.name}`);
      res = await createLibrary(lib.name, lib.description, code, lib.language);
      
```